s2s/time.py

```diff
@@ -417,38 +421,106 @@ def __repr__(self):
+            >>> calendar.map_years(2020, 2021, flat=True)
+            0    (2021-04-18, 2021-10-15]
+            1    (2020-10-20, 2021-04-18]
+            2    (2020-04-18, 2020-10-15]
+            3    (2019-10-21, 2020-04-18]
+            dtype: interval
+            
+            >>> calendar.set_traintest_method("kfold", n_splits = 2)
+            >>> calendar.get_traintest()
+               anchor_year  i_intervals                 intervals fold_0 fold_1
+            0         2021            0  (2021-04-18, 2021-10-15]  train   test
+            1         2021            1  (2020-10-20, 2021-04-18]  train   test
+            2         2020            0  (2020-04-18, 2020-10-15]   test  train
+            3         2020            1  (2019-10-21, 2020-04-18]   test  train
+        """
+        # checker if generated intervals are flat.
+        if self.intervals.ndim != 1:
+            raise ValueError("Please set `flat = True` when calling `map_years` or `map_data`")
+
+        # infer anchor years from give intervals
+        anchor_years = [timestamp.right.year for timestamp in self.intervals[::self._n_intervals]]
+        # Create DataFrame from intervals Series to store the train/test groups
+        traintest_base = pd.DataFrame(data = {
+        'anchor_year': np.repeat(anchor_years, self._n_intervals),
+        'i_intervals': np.tile(range(self._n_intervals), len(anchor_years)),
+        'intervals': self.intervals,
+        })
+
+        # checker if the method is configured.
+        if self._traintest_method is not None:
+            self.traintest = traintest.ALL_METHODS[self._traintest_method](traintest_base,
+                **self._method_kwargs)
+
+        return self.traintest
+
+    def set_traintest_method(self, method: str, **method_kwargs: Optional[dict]):
+        """
+        The user must choose a method here. And the method will be used by
+        all traintest splitting methods.
+
+        Args:
+            method: one of the methods available in `s2s.traintest`
+            method_kwargs: keyword arguments that will be passed to `method`
+        """
+        # checker if the given method is supported in `s2s.traintest`
+        if method not in [key for key, _ in traintest.ALL_METHODS.items()]:
+            raise ValueError("The given method is not supported by `s2s.traintest`.")
+        self._traintest_method = method
+        self._method_kwargs = method_kwargs
+
+    def get_train(self) -> pd.DataFrame:
+        """Return indices for training data indices using given strategy.
+        """
         raise NotImplementedError
 
-    def get_train_test_indices(self, strategy, params):  # noqa
-        """Shorthand for getting both train and test indices."""
-        # train = self.get_train_indices()
-        # test = self.get_test_indices()
+    def get_test(self) -> pd.DataFrame:
+        """Return indices for test data indices using given strategy.
+        """
```

Comment:
Why are these still not implemented?

Reply:
Ah, I guess it's ambiguous what to return, e.g. for K-fold when you have multiple folds. Need to discuss what behavious we expect.

Reply:
Yes, that's exactly what I was thinking. 😉